tests/Yaapii.Xml.Xambly.Tests/Directive/DirectivesTest.cs

```diff
@@ -288,6 +288,31 @@ public void ParsesGrammar()
         }
 
 
+        [Fact]
+        public void UsefulInfoAtAddingAttributeToDocumentNode()
+        {
+            try
+            {
+                var xml = new XmlDocument();
+                var xambler =
+                    new Xambler(
+                        new Directives()
+                            .Add("root")
+                            .Add("child")
+                            .Xpath("/")
+                            .Attr("some", "attribute")
+                    ).Apply(xml);
+                Assert.True(false, "Exception expected!");
```

Comment:
Please use `Assert.Throws` from xunit framework.

Reply:
`Assert.Throws` doesn't give the possibility to checks exception's message. If a implement that I can check the existence of an exception but not it's content.
I would prefer to let it as it is.

Reply:
Maybe we should throw a more typed exception? Relying on the message content of an exception fells kind da wrong to me. What do you think @Meerownymous ?

Reply:
Agree with @koeeenig, it should be tested if the correct Exception type is throw . If you want to check the message, this is not the right place - that should be done in a test which tests the Exception class. message is for context, type is for application flow.

Reply:
Okay, I will change it.
But an open question for me: How test the exception message?
@Meerownymous: You say: "that should be done in a test which tests the Exception class"
The message is generated in an class which uses the exception (here `Directives()`). Then I haven't the chance to test the message content in an exception test.
Do we need an new different `Assert.Throws()` method which tests the content of an exception message?

Reply:
I simply think, testing the message for each class which rises the exception is kind of over-engineered. Lets test the type, and maybe on the other hand if the exception class is built correctly, but thats it.

Reply:
okay